node/resolvers/profile/services.ts

```diff
@@ -62,18 +62,26 @@ export function getAddresses(context: Context, currentUserProfile?: Profile) {
 
 export async function getPayments(context: Context) {
   const {
-    clients: { profile },
+    clients: { profile, licenseManagerExtended },
     vtex: { currentProfile },
   } = context
 
+  const { PIIEnabled } = await licenseManagerExtended.getCurrentAccount()
   const paymentsRawData = await profile.getUserPayments(currentProfile, context)
+  const isPaymentDataEmpty = PIIEnabled
+    ? !paymentsRawData[0].document.paymentData
+    : !paymentsRawData?.paymentData
```

Comment:
Sem saber exatamente o formato dos dados que são retornados aqui, mas o que acha de tentar reduzir a lógica de `isPaymentDataEmpty` e `paymentData`?

Acha que algo assim faz sentido?
```suggestion
  const paymentData = PIIEnabled
    ? paymentsRawData[0].document.paymentData
    : JSON.parse(paymentsRawData.paymentData)
```

Daí o `if` poderia ser um `if (!paymentData)` e o bloco de `paymentData` abaixo poderia ser removido.